dpp/src/identity/identity_public_key.rs

```diff
@@ -325,29 +320,13 @@ impl IdentityPublicKey {
             return Ok(self.data.clone());
         }
 
-        let original_key = match self.data.len() {
-            65 => {
-                let public_key = vec::vec_to_array::<65>(&self.data)
-                    .map_err(|e| ProtocolError::ParsingError(e.to_string()))?;
-
-                PublicKey::from_slice(&public_key)
-                    .map_err(|e| anyhow!("unable to create pub key - {}", e))?
-            }
-
-            33 => {
-                let public_key = vec::vec_to_array::<33>(&self.data)
-                    .map_err(|e| ProtocolError::ParsingError(e.to_string()))?;
-                PublicKey::from_slice(&public_key)
-                    .map_err(|e| anyhow!("unable to create pub key - {}", e))?
-            }
-            _ => {
-                return Err(ProtocolError::ParsingError(format!(
-                    "the key length is invalid: {} Allowed sizes: 33 or 65 bytes",
-                    self.data.len()
-                )));
-            }
-        };
-        Ok(original_key.pubkey_hash().to_vec())
+        match self.data.len() {
```

Comment:
Shouldn't it be based on type not on size?

Reply:
Fixed